federation-2/supergraph/src/command/compose.rs

```diff
@@ -30,10 +32,41 @@ impl Compose {
     }
 
     fn do_compose(&self) -> BuildResult {
-        let supergraph_config = SupergraphConfig::new_from_yaml_file(&self.config_file)?;
+        let buffer = match &self.config_file {
+            None => {
+                let mut buffer = String::new();
+                std::io::stdin()
+                    .read_to_string(&mut buffer)
+                    .map_err(|e| -> ConfigError {
+                        ConfigError::InvalidConfiguration {
+                            message: e.to_string(),
+                        }
+                    })?;
+                Ok(buffer)
+            }
+            Some(config_path) => {
+                if config_path.exists() {
+                    let contents = std::fs::read_to_string(config_path).map_err(|e| {
+                        ConfigError::MissingFile {
+                            file_path: config_path.to_string(),
+                            message: e.to_string(),
+                        }
+                    })?;
+                    Ok(contents)
+                } else {
+                    Err(ConfigError::MissingFile {
+                        file_path: config_path.to_string(),
+                        message: "Unable to find supergraph.yaml file".to_owned(),
```

Comment:
would want to use the real file name (or say it came from stdin) here instead of `supergraph.yaml`